### PR TITLE
chore: fix table ajax demo

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      x     |
| 🇨🇳 Chinese |     x      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/table/demo/ajax.tsx [hunk 1]
```diff
@@ -84,15 +84,11 @@ const App: React.FC = () => {
   };
 
   useEffect(() => {
-    fetchData();
-  }, [tableParams.pagination?.current, tableParams.pagination?.pageSize]);
+    fetchData(tableParams);
+  }, []);
```
Comment (afc163) [inline components/table/demo/ajax.tsx:88]: 统一用副作用去触发请求，而不是分开发 fetch。

Comment (afc163) [reply components/table/demo/ajax.tsx:88]: 这里要改的是 deps 条件。

File: components/table/demo/ajax.tsx [hunk 2]
```diff
@@ -85,13 +86,20 @@ const App: React.FC = () => {
 
   useEffect(() => {
     fetchData();
-  }, [tableParams.pagination?.current, tableParams.pagination?.pageSize]);
+  }, [
+    tableParams.pagination?.current,
+    tableParams.pagination?.pageSize,
+    tableParams?.sortOrder,
+    tableParams?.sortField,
+  ]);
```
Comment (afc163) [inline components/table/demo/ajax.tsx:94]: filters 的变化也加上？

File: components/table/demo/ajax.tsx [hunk 3]
```diff
@@ -85,13 +86,21 @@ const App: React.FC = () => {
...
+  }, [
+    tableParams.pagination?.current,
+    tableParams.pagination?.pageSize,
+    tableParams?.sortOrder,
+    tableParams?.sortField,
+    tableParams.filters,
+  ]);
 
   const handleTableChange: TableProps['onChange'] = (pagination, filters, sorter) => {
+    console.log('sorter', sorter);
```
Comment (afc163) [inline components/table/demo/ajax.tsx:98]: 这个应该去掉。

File: components/table/demo/ajax.tsx [hunk 4]
```diff
@@ -91,6 +91,7 @@ const App: React.FC = () => {
     tableParams.pagination?.pageSize,
     tableParams?.sortOrder,
     tableParams?.sortField,
+    tableParams.filters,
```
Comment (afc163) [inline components/table/demo/ajax.tsx:94]: 需要时 primitive 类型。

Comment (crazyair) [reply components/table/demo/ajax.tsx:94]: 这个就是 gender 了

File: components/table/demo/ajax.tsx [hunk 5]
```diff
@@ -85,13 +86,20 @@ const App: React.FC = () => {
 
   useEffect(() => {
     fetchData();
-  }, [tableParams.pagination?.current, tableParams.pagination?.pageSize]);
+  }, [
+    tableParams.pagination?.current,
+    tableParams.pagination?.pageSize,
+    tableParams?.sortOrder,
+    tableParams?.sortField,
+    tableParams.filters,
```
Comment (afc163) [inline components/table/demo/ajax.tsx:94]:
```suggestion
    JSON.stringify(tableParams.filters),
```

File: components/table/demo/ajax.tsx [hunk 6]
```diff
@@ -85,13 +86,20 @@ const App: React.FC = () => {
...
+    JSON.stringify(tableParams.filters),
+  ]);
 
   const handleTableChange: TableProps['onChange'] = (pagination, filters, sorter) => {
     setTableParams({
       pagination,
       filters,
-      ...sorter,
+      sortOrder: Array.isArray(sorter) ? undefined : sorter.order,
+      sortField: Array.isArray(sorter) ? undefined : sorter.field,
```
Comment (afc163) [inline components/table/demo/ajax.tsx:102]: 为啥 sorter 是 array 就不传？

**Review by afc163 — APPROVED** [review]

**Pull request merged**